lightning/mydump/loader.go

```diff
@@ -264,6 +279,45 @@ func (l *MDLoader) shouldSkip(table *filter.Table) bool {
 	return len(l.filter.ApplyOn([]*filter.Table{table})) == 0
 }
 
+func (s *mdLoaderSetup) route() error {
+	r := s.loader.router
+	if r == nil {
+		return nil
+	}
+
+	knownDBNames := make(map[string]string)
+	for _, info := range s.dbSchemas {
+		knownDBNames[info.tableName.Schema] = info.path
+	}
+
+	run := func(arr []fileInfo) error {
+		for i, info := range arr {
+			dbName, tableName, err := r.Route(info.tableName.Schema, info.tableName.Name)
+			if err != nil {
+				return errors.Trace(err)
+			}
+			if _, ok := knownDBNames[dbName]; !ok {
+				path := knownDBNames[info.tableName.Schema]
+				knownDBNames[dbName] = path
+				s.dbSchemas = append(s.dbSchemas, fileInfo{
```

Comment:
if we route database `A` to database `B`, is an empty database `A` still created?

Reply:
Fixed.